scipy/signal/tests/test_upfirdn.py

```diff
@@ -228,4 +228,8 @@ def test_modes(self, size, h_len, mode, dtype):
         ypad = upfirdn(h, xpad, up=1, down=1, mode='constant')
         y_expected = ypad[npad:-npad]
 
-        assert_allclose(y, y_expected, atol=1e-6, rtol=1e-6)
+        if dtype in (np.float32, np.complex64):
+            atol = rtol = 1e-5
+        else:
+            atol = rtol = 1e-12
+        assert_allclose(y, y_expected, atol=atol, rtol=rtol)
```

Comment:
Would something like `atol = rtol = np.finfo(dtype).eps * 1e2` work?
```
>>> np.finfo(np.complex64).eps * 1e2
1.1920928955078125e-05
>>> np.finfo(np.complex128).eps * 1e2
2.220446049250313e-14
```
It cuts down on lines and seems more principled, but if it doesn't work then what you have seems fine

Reply:
I meant the differences. The values are in the same order of magnitude as far as I can see. `rtol` is espcially helpful when the compared quantities have order of magnitude drastically changing.

Reply:
Okay.  I am hesitant to make it tighter than it currently is as I cannot reproduce the failures seen on scipy-wheels when testing locally (even `atol = rtol = 0` passes on my system).